### PR TITLE
DOC: Improve documentation of `isscalar`.

### PR DESCRIPTION
**Pull request by raamana**
Following the discussion on this [twitter thread](https://twitter.com/raamana_/status/1162746638605176832?s=20), this PR tries to improve the docs for `np.isscalar()`

Please edit as you feel is ideal for this merge.

PS: tried to add the word DOC to the commit messages, it wasn't working in my IDE for some reason.

### CODE REVIEW

**Comment by mattip** [comment]
Perhaps `isinstance(element, np.generic)` would be a better workaround?

File: numpy/core/numeric.py [hunk 1]
```diff
@@ -1772,30 +1772,34 @@ def _frombuffer(buf, dtype, shape, order):
 
 
 @set_module('numpy')
-def isscalar(num):
+def isscalar(element):
```
Comment (eric-wieser) [inline numpy/core/numeric.py]: Are we ok with renaming this argument? I think the chance of it breaking anyone is ~0.

Comment (mattip) [reply numpy/core/numeric.py]: it seems harmless

**Comment by raamana** [comment]
Thanks @mattip and @eric-wieser for the review and merging it. Super glad I was able to contribute to the mighty numpy project, even though this is such a tiny contrib :)

**Pull request merged**